avocado/utils/network/interfaces.py

```diff
@@ -388,52 +400,92 @@ def save(self, ipaddr, netmask):
         current_distro = distro_detect()
 
         filename = f"ifcfg-{self.name}"
+        prefix = self.netmask_to_cidr(netmask)
         if current_distro.name in ["rhel", "fedora"]:
-            path = "/etc/sysconfig/network-scripts"
+            if(current_distro.name == 'rhel' and current_distro.version == '9'):
+                filename = f"{self.name}.nmconnection"
+                path = "/etc/NetworkManager/system-connections"
+            else:
+                path = "/etc/sysconfig/network-scripts"
         elif current_distro.name == "SuSE":
             path = "/etc/sysconfig/network"
         else:
             msg = "Distro not supported by API. Could not save ipaddr."
             raise NWException(msg)
 
-        ifcfg_dict = {
-            "TYPE": self.if_type,
-            "BOOTPROTO": "static",
-            "NAME": self.name,
-            "DEVICE": self.name,
-            "ONBOOT": "yes",
-            "IPADDR": ipaddr,
-            "NETMASK": netmask,
-            "IPV6INIT": "yes",
-            "IPV6_AUTOCONF": "yes",
-            "IPV6_DEFROUTE": "yes",
-        }
-        if current_distro.name == "SuSE":
-            ifcfg_dict.pop("BOOTPROTO")
+        if(current_distro.name == 'rhel' and current_distro.version == '9'):
+            ifcfg_dict = ''
+            if os.path.exists(f"{path}/{filename}"):
+                pass
```

Comment:
when condition is true and there is nothing to be done .. than why not just check for `if not`  and move else part here